src/util/Location.ts

```diff
@@ -45,10 +45,31 @@ export class Location {
         return fs.ensureDir(this.basePath);
     }
 
-    public async unlinkIfExists(): Promise<void> {
-        if (await this.exists()) {
-            await fs.unlink(this.basePath);
-        }
+    /**
+     * Remove a path, retrying a few times after a delay in case of error.
+     * If the path does not exist, silently does nothing.
+     */
+    public async remove(maxRetries = 3, retryDelay = 1): Promise<void> {
+        let currRetryNumber = 0;
+        while (true) {
+            if (currRetryNumber >= maxRetries) {
+                // Don't catch errors
+                await fs.remove(this.basePath);
+            } else {
+                // Try again in a few seconds in case of errors
+                try {
+                    await fs.remove(this.basePath);
+                } catch {
+                    currRetryNumber += 1;
+                    await new Promise(
+                        (resolve) => setTimeout(resolve, retryDelay * 1000)
+                    );
+                    continue;
+                }
+            }
+            // Success
+            return;
```

Comment:
unnecessary return stmt

Reply:
But it's a `while true`?

Reply:
I tried rewriting to a `do { .. } while (false)`, but I find it more confusing.

Reply:
sorry I'm an idiot. I didn't see that `return;` is in the scope of the `while true`.
Your comment already made it clearer